### DISCUSSION TITLE
Is there a way to manually define game installations?

### DISCUSSION THREAD

**Discussion by TheOverpassArsonist** [ebkr/r2modmanPlus#1516] [Q&A]
I noticed when I was adding a game to this that after you select a game you're given two options to select which store manages your game, but is there any way to say "none" here and input game details manually? For instance if you install a game via something like Lutris on linux it won't be managed by any stores, yet it should still be perfectly moddable. (note : I did search for similar discussions here, but the closest I could find was this #737 which seems unrelated, but it's possible it's a matter of phrasing and a similar discussion does exist that I just couldn't find.)

**Comment by ebkr** [comment]
No there isn't.

It doesn't particularly make sense because the point is that the manager pulls from Thunderstore to help make things easier. Manually adding a game and locally importing every single mod you want to use is tedious and at that point you'd be better off with a manual install and using the launch parameters yourself. The only benefit you'd get out of this is by not having to specify the launch parameters yourself.

The other issue is that it provides a workaround for launching on pirated copies. Ideally we don't allow this because:
1. Better abides by community server rules, which helps to improve relations between the modding communities and the game developers
2. Prevents error reports on non-official builds

**Reply by TheOverpassArsonist** [reply]
I'm talking about manually specifying a game installation, not every individual mod. I have several games that I own on other platforms (notably GOG) that are just distributed as standalone installers. (or can be, GOG has their own store interface and there are third party ones like Heroic as well but for some games I'm more partial to using external managers like Lutris since it gives me more options for tweaking the prefix and how they're ran.)

For these games (ex : [Against The Storm](https://www.gog.com/en/game/against_the_storm) is visible in the manager and can be purchased on GOG, 50% off right now too.) not being able to manually specify a game directory means I can't use the tool to mod the game because of where/how I bought it. (or got it from something like Twitch Prime's free game rewards, since a lot of those are GOG codes.) Trying to fight piracy by arbitrarily restricting what installation methods players can use isn't going to actually stop pirates but it's definitely going to screw over people who just prefer to not overly rely on tightly controlled storefronts. 

At the risk of overstepping here, I really don't even see how anyone could see something to complain about with this being used for pirated games anyway. This is a third party mod manager, it doesn't enable piracy or even have anything to do with how the game is aquired, all it needs to do is enable/disable mods. The most I could see this ever mattering would be if someone released a mod that allowed you to play a pirated copy of the game, but such a 'mod' would require you already have the gamefiles anyway, at which point it'd just be distributed as a cracked copy of the game, not a mod that requires you already have the legit gamefiles. (and, even in that case, the mod manager isn't responsible for the mods people make; no-one is blaming Forge or GDLauncher for a Minecraft sex mod) I just really don't see how people would even associate the two concepts together to get mad about it in the first place. 

Stopping error reports from people using non-official builds of games I can sorta see, but I'm not even sure this would actually stop that. Couldn't someone just overwrite the game files with the unofficial build and then run it? Unless this tool automatically runs a steam validation check or something constantly, someone could just overwrite the game files in the steam game directory and then it'd launch it just fine with the unofficial build. I don't recall when specifically I had to do this, but I do vaguely recall fooling steam like this before to make it launch an intermediate loader or something instead of the game itself. (If I had to guess it was probably making Steam launch the alternative XCOM2 mod launcher instead of the normal 2K launcher window thing, but I could be misattributing that vague memory from another game)

Also, and I haven't tested this, (currently on a secondary windows laptop) but I feel like this requirement to use official storefronts like Steam could also introduce another issue becuase on my main Linux desktop I have Steam installed as a flatpak with minimal permissions. Permissions which, among other things, prevent the SteamAPI from working. From how this tool seems to work though, I'd guess it uses the SteamAPI to prompt steam to run the game and such, which means that if you have an installation where the SteamAPI isn't accessible this tool wouldn't work either. (again, have not tested this) 

Additionally, from how you talk about it here it sounds like how this works under the hood is that it tells steam to run the game with certain launch parameters which enable the mods instead of modifying the game files directly, but in that case being able to specify manual installation directories seems like it'd also let you do something else really cool (at least on linux) in that you could use it to generate launch commands and then create seperate .desktop files to launch the same game with different profiles applied. (granted, that's a biiit outside of the immediate scope here, but it's tangentially related because to my knowledge there isn't any easy way of doing that via Steam itself, or any other storefront to my knowledge)

In general this seems like an incredibly limiting and seemingly arbitrary decision.
